src/tb_client_exports.zig

```diff
@@ -28,13 +29,23 @@ comptime {
 
 fn init(
     out_client: *tb.tb_client_t,
-    cluster_id: u128,
+    cluster_id_ptr: *const [16]u8,
     addresses_ptr: [*:0]const u8,
     addresses_len: u32,
     on_completion_ctx: usize,
     on_completion_fn: tb.tb_completion_t,
 ) callconv(.C) tb.tb_status_t {
     const addresses = @as([*]const u8, @ptrCast(addresses_ptr))[0..addresses_len];
+
+    // Passing u128 by value is prone to ABI issues. Pass as a [16]u8, and explicitly copy into
+    // memory we know will be aligned correctly.
+    const cluster_id: u128 = blk: {
+        var cluster_id: u128 = undefined;
+        stdx.copy_disjoint(.exact, u8, std.mem.asBytes(&cluster_id), cluster_id_ptr);
+
+        break :blk cluster_id;
+    };
```

Comment:
```suggestion
    const cluster_id = std.mem.bytesToValue(u128, cluster_id_ptr);
```

Reply:
I thought about `bytesToValue`, but isn't there potential for an alignment issue here? (I'm not sure!)

Reply:
Looking at the std lib implementation, it just cast and dereference the pointer... I doubt it'll cause any issues with alignment (IIUC the issue was caused in the argument list when crossing the FFI). But also I'm not sure 😰

Is there a way to validate this? At least we could add a comment explaining why we avoid bytesToValue (maybe a good candidate for stdx).

Reply:
> IIUC the issue was caused in the argument list when crossing the FFI

Yep, it was an assembly level / ABI issue.

> Is there a way to validate this? At least we could add a comment explaining why we avoid bytesToValue (maybe a good candidate for stdx).

So, `bytesToValue` calls and uses `bytesAsValue`, which does keep the pointer alignment, so it's accessing and copying from a `*align(1) const u128`.

I think this could cause unaligned accesses - which are seemingly handled both on x86_64 and arm64, but maybe it would be better to avoid it entirely...